### PR TITLE
Adds mirrored images for bci-micro and bci-busybox.

### PR DESCRIPTION
**Pull request by KevinJoiner**
Issue: https://github.com/rancher/rancher/issues/35587

### CODE REVIEW

**Review by jiaqiluo — CHANGES_REQUESTED** [review]
~Because [registry.rancher.com](http://registry.rancher.com/) is an alias of [registry.suse.com](http://registry.suse.com/), what we need to do is replace `registry.suse.com` with `registry.rancher.com`~ 

Please ignore this comment. We still need to mirror those images, so the changes in the PR are valid.

File: pkg/image/utilities/utilities_test.go [hunk 1]
```diff
@@ -19,11 +19,11 @@ func TestCheckImage(t *testing.T) {
 	for k, v := range imageListAndErrorExpectations {
 		err := checkImage(k)
 		if err != nil && !v {
-			t.Logf("expected error when checking image %s", k)
+			t.Logf("did not expected error when checking image %s", k)
```
Comment (cbron) [inline pkg/image/utilities/utilities_test.go:22]:
```suggestion
			t.Logf("did not expect error when checking image %s", k)
```

File: pkg/image/utilities/utilities_test.go [hunk 2]
```diff
@@ -19,11 +19,11 @@ func TestCheckImage(t *testing.T) {
...
 			t.Fail()
 		}
 		if err == nil && v {
-			t.Logf("did not expect error when checking image %s", k)
+			t.Logf("expect error when checking image %s", k)
```
Comment (cbron) [inline pkg/image/utilities/utilities_test.go:26]:
```suggestion
			t.Logf("expected error when checking image %s", k)
```

**Review by cbron — COMMENTED** [review]
two nits, and needs images mirrored, then good

File: pkg/image/resolve.go [hunk 3]
```diff
@@ -130,8 +130,8 @@ func setRequirementImages(osType OSType, imagesSet map[string]map[string]struct{
 	case Linux:
 		addSourceToImage(imagesSet, settings.ShellImage.Get(), coreLabel)
 		addSourceToImage(imagesSet, settings.MachineProvisionImage.Get(), coreLabel)
-		addSourceToImage(imagesSet, "registry.suse.com/bci/bci-busybox:15.4.11.2", coreLabel)
-		addSourceToImage(imagesSet, "registry.suse.com/bci/bci-micro:15.4.14.3", coreLabel)
+		addSourceToImage(imagesSet, "rancher/mirrored-bci-busybox:15.4.11.2", coreLabel)
+		addSourceToImage(imagesSet, "rancher/mirrored-bci-micro:15.4.14.3", coreLabel)
```
Comment (cbron) [inline pkg/image/resolve.go:134]: both of these exist now

**Review by cbron — APPROVED** [review]

**Review by jiaqiluo — APPROVED** [review]

**Review by cbron — APPROVED** [review]

**Pull request merged**